internal/host/validator.go

```diff
@@ -614,3 +626,32 @@ func isInvalidImageStatus(imageStatus *models.ContainerImageAvailability) bool {
 	return imageStatus.Result == models.ContainerImageAvailabilityResultFailure ||
 		(imageStatus.SizeBytes > 0 && imageStatus.DownloadRate < ImageStatusDownloadRateThreshold)
 }
+
+/*
+   This is a pre-install validation that checks that the boot device was either not tested for sufficient disk speed
+   or the disk speed check has been successful.  Since disk speed test is performed after installation has started,
+   in order to have result for such test, the result has to be from a previous installation attempt.
+   Since all pre-install validations have to pass before starting installation, it is mandatory that in case installation
+   on the current boot device has not been attempted yet, this validation must pass.
+*/
+func (v *validator) sufficientOrUnknownInstallationDiskSpeed(c *validationContext) ValidationStatus {
+	info, err := v.getBootDeviceInfo(c.host)
+	if err != nil {
+		return ValidationError
+	}
+
+	return boolValue(info == nil || info.DiskSpeed == nil || !info.DiskSpeed.Tested || info.DiskSpeed.ExitCode == 0)
+}
+
+func (v *validator) printSufficientOrUnknownInstallationDiskSpeed(c *validationContext, status ValidationStatus) string {
+	switch status {
+	case ValidationSuccess:
+		return "Sufficient disk speed"
```

Comment:
Can we print something else if the test hasn't run yet?

Reply:
Done